controllers/release/release_adapter.go

```diff
@@ -47,24 +48,26 @@ import (
 
 // Adapter holds the objects needed to reconcile a Release.
 type Adapter struct {
-	release *v1alpha1.Release
-	logger  logr.Logger
 	client  client.Client
-	context context.Context
+	ctx     context.Context
+	loader  loader.ObjectLoader
+	logger  logr.Logger
+	release *v1alpha1.Release
 	syncer  *syncer.Syncer
 }
 
 // finalizerName is the finalizer name to be added to the Releases
 const finalizerName string = "appstudio.redhat.com/release-finalizer"
 
 // NewAdapter creates and returns an Adapter instance.
-func NewAdapter(release *v1alpha1.Release, logger logr.Logger, client client.Client, context context.Context) *Adapter {
+func NewAdapter(ctx context.Context, client client.Client, release *v1alpha1.Release, loader loader.ObjectLoader, logger logr.Logger) *Adapter {
 	return &Adapter{
-		release: release,
-		logger:  logger,
 		client:  client,
-		context: context,
-		syncer:  syncer.NewSyncerWithContext(client, logger, context),
+		ctx:     ctx,
+		loader:  loader,
```

Comment:
Why again do we need to do `a.loader.func` and have it be part of the adapter instead of `loader.func`?

Reply:
Because that way we can create an adapter which operations use a mock.